dot/state/block.go

```diff
@@ -291,35 +247,73 @@ func (bs *BlockState) SetHeader(header *types.Header) error {
 	return nil
 }
 
-// SetBlock will add a block to the DB
-func (bs *BlockState) SetBlock(block *types.Block) error {
-	// Add the blockHeader to the DB
-	err := bs.SetHeader(block.Header)
+// GetBlockBody will return Body for a given hash
+func (bs *BlockState) GetBlockBody(hash common.Hash) (*optional.Body, error) {
```

Comment:
don't return an optional here, if it doesn't exist then an error will be returned

Reply:
That was the previous logic in the code, I left it like that to avoid changing things out of the scope in the ticket. But if you want I can get rid of the optionals, no problem.

Reply:
updated.